addons/sale/models/onboarding_step.py

```diff
@@ -0,0 +1,89 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+import base64
+
+from odoo import _, api, Command, models
+from odoo.modules.module import get_resource_path
+from odoo.tools import file_open
+
+
+class OnboardingStep(models.Model):
+    _inherit = 'onboarding.onboarding.step'
+
+    @api.model
+    def action_open_sale_onboarding_payment_provider(self):
+        self.env.company.get_chart_of_accounts_or_fail()
+        return self.env['ir.actions.actions']._for_xml_id('sale.action_open_sale_payment_provider_onboarding_wizard')
+
+    @api.model
+    def _get_sample_sales_order(self):
+        """ Get a sample quotation or create one if it does not exist. """
+        # use current user as partner
+        partner = self.env.user.partner_id
+        company_id = self.env.company.id
+        # is there already one?
+        sample_sales_order = self.env['sale.order'].search([
+            ('company_id', '=', company_id),
+            ('partner_id', '=', partner.id),
+            ('state', '=', 'draft'),
+        ], limit=1)
+        if not sample_sales_order:
+            # take any existing product or create one
+            product = self.env['product.product'].search([], limit=1)
+            if not product:
+                default_image_path = get_resource_path('product', 'static/img', 'product_product_13-image.jpg')
+                with file_open(default_image_path, 'rb') as default_image_stream:
+                    product = self.env['product.product'].create({
+                        'name': _('Sample Product'),
+                        'active': False,
+                        'image_1920': base64.b64encode(default_image_stream.read())
+                    })
+                product.product_tmpl_id.active = False
+            sample_sales_order = self.env['sale.order'].create({
+                'partner_id': partner.id,
+                'order_line': [
+                    Command.create({
+                        'name': _('Sample Order Line'),
+                        'product_id': product.id,
+                        'product_uom_qty': 10,
+                        'price_unit': 123,
+                    })
+                ]
+            })
+        return sample_sales_order
+
+    @api.model
+    def action_open_sale_onboarding_sample_quotation(self):
+        """ Onboarding step for sending a sample quotation. Open a window to compose an email,
+            with the edi_invoice_template message loaded by default. """
+        sample_sales_order = self._get_sample_sales_order()
+        template = self.env.ref('sale.email_template_edi_sale', False)
+
+        self.env['mail.compose.message'].with_context(
+            mark_so_as_sent=True,
+            default_email_layout_xmlid='mail.mail_notification_layout_with_responsible_signature',
+            proforma=self.env.context.get('proforma', False),
+            force_email=True,
```

Comment:
This key is probably unnecessary (I think it is used to raise if we create a partner without email)

Reply:
Indeed.